### PR TITLE
Recent snippets

### PR DESCRIPTION
**Pull request by ikalnytskyi**
(see details in commit messages)

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const setRecentSnippets = snippets => ({
+  type: 'SET_RECENT_SNIPPETS',
+  snippets,
+});
+
+export const fetchRecentSnippets = dispatch => (
+  fetch('http://api.xsnippet.org/snippets')
```
Comment (malor) [inline src/actions/index.js:7]: We probably want this to be configurable, so that e.g. we can run the app locally.

**Review by lotrien — APPROVED** [review]

File: src/reducers/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { List, Map } from 'immutable';
+import { combineReducers } from 'redux';
+
+const snippets = (state = Map(), action) => {
```
Comment (malor) [inline src/reducers/index.js:4]: Why do we need both `snippets` and `recent` reducers that correspond to the very same action `SET_RECENT_SNIPPETS `?

**Review by malor — APPROVED** [review]
TBH, I did not get all of it, but looks good to go and seems to be working correctly :)

File: src/actions/index.js (hunk 1 above)
Comment (ikalnytskyi) [reply src/actions/index.js:7]: You are right, I just have no idea how to implement settings in JS. :D

File: src/reducers/index.js (hunk 2 above)
Comment (ikalnytskyi) [reply src/reducers/index.js:4]: That's the good question. In Redux, the Store can have only one reducer that receives action and somehow updates the state. However, this fact usually means that you will have one helluva reducer that own everything and updates everything.

So they provide a util function that allows you to split reducer logic across multiple reducers and then combine them into one. The one, that you will pass to the Store (see combineReducers below).

So, for instance, in our case we want to have the following store:

```javascript
{
  "snippets": {
    1: {
      "id": 1,
      "title": "Untitled",
      "content": "bla=bla",
      "syntax": "JavaScript",
    },
    2: {
      "id": 2,
      "title": "Something",
      "content": "magic",
      "syntax": "Python",
    },
  },
  "recent": [1, 2],
}
```

And following Redux best practices, we need to have one reducer for each node, i.e. one for `snippets` node and one for `recent` node. It's required, cause `combineReducer` function will only partial state for each income action, i.e. `snippet` reducer will take `state.snippets` as `state` and `recent` reducer will receive `state.recent` as `state`.

Comment (malor) [reply src/reducers/index.js:4]: Thanks for redux 101! 👍

**Pull request merged**